libusb/libusb.h

```diff
@@ -1927,10 +1927,18 @@ static inline void libusb_fill_control_transfer(
 	transfer->endpoint = 0;
 	transfer->type = LIBUSB_TRANSFER_TYPE_CONTROL;
 	transfer->timeout = timeout;
-	transfer->buffer = buffer;
 	if (setup)
-		transfer->length = (int) (LIBUSB_CONTROL_SETUP_SIZE
+	{
+		int length = (int) (LIBUSB_CONTROL_SETUP_SIZE
```

Comment:
Does it also require the helper variable?

Reply:
IIRC, yes in this case. Because the compiler needs to see both the pointer field (buffer) and the size field (length) updated back to back with no intervening stuff.

https://clang.llvm.org/docs/BoundsSafety.html#maintaining-correctness-of-bounds-annotations

"Additionally, an explicit bounds annotation such as `int *__counted_by(count) buf` defines a relationship between two variables, buf and count: namely, that buf has count number of elements available. This relationship must hold even after any of these related variables are updated. To this end, the model requires that assignments to buf and count must be side by side, with no side effects between them. This prevents buf and count from temporarily falling out of sync due to updates happening at a distance."